backend/pkg/exporter/modules/execution_deposits_exporter.go

```diff
@@ -145,89 +158,115 @@ func (d *executionDepositsExporter) OnChainReorg(event *constypes.StandardEventC
 	return nil // nop
 }
 
-// can take however long it wants to run, is run in a separate goroutine, so no need to worry about blocking
 func (d *executionDepositsExporter) OnFinalizedCheckpoint(event *constypes.StandardFinalizedCheckpointResponse) (err error) {
-	// important: have to fetch the actual finalized epoch because even tho its called on finalized checkpoint it actually emits for each justified epoch
-	// so we have to do an extra request to get the actual latest finalized epoch
-	res, err := d.CL.GetFinalityCheckpoints("head")
-	if err != nil {
-		return err
-	}
+	return nil // nop
+}
 
+func (d *executionDepositsExporter) OnHead(event *constypes.StandardEventHeadResponse) (err error) {
 	var nearestELBlock sql.NullInt64
-	err = db.ReaderDb.Get(&nearestELBlock, "select exec_block_number from blocks where slot <= $1 and exec_block_number > 0 order by slot desc limit 1", res.Data.Finalized.Epoch*utils.Config.Chain.ClConfig.SlotsPerEpoch)
+	err = db.ReaderDb.Get(&nearestELBlock, "select exec_block_number from blocks where exec_block_number > 0 order by slot desc limit 1")
```

Comment:
this is really meh. it made a bit more sense when we were following the finalized checkpoints (we had to resolve the slot to a exec block somehow) but even that wasnt that clean (you can resolve slot => exec block num using [`/eth/v2/beacon/blocks/{block_id}`](https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockV2) `data.message.body.execution_payload.block_number` )

i would rather 
- listen to block events from erigon by adding a new `OnExecutionHead` listener to the modular exporter system
- resolve the slot to a exec block hash using the beacon api as described above
- do this query in the actual exporting loop below (makes no difference doing it here or there, besides having to store it variable when doing the former), though I'm not a big fan of having exporters depend on other exporters because this makes triaging hard and can cause compounding delays